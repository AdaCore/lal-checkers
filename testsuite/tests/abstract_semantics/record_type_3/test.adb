procedure Test is
   type Point is record
      x : Integer;
      y : Integer;
   end record;
   type Rectangle is record
      top_left: Point;
      bottom_right: Point;
   end record;
   r : Rectangle;
   i : Integer := 12;
begin
   r.top_left.x := 12;
   r.top_left.y := i - 2;
   r.bottom_right.x := i + 5;
   r.bottom_right.y := 22;
end Ex1;
