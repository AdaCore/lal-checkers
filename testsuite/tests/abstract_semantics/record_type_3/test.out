{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "i": "[-2147483648, 2147483647]", 
          "r": "(([-2147483648, 2147483647], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([-2147483648, 2147483647], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [10, 10]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [10, 10]), ([17, 17], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assign5", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [10, 10]), ([17, 17], [22, 22]))"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "i": "[-2147483648, 2147483647]", 
          "r": "(([-2147483648, 2147483647], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ]
  }, 
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "i": "[-2147483648, 2147483647]", 
          "r": "(([-2147483648, 2147483647], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([-2147483648, 2147483647], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [10, 10]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [10, 10]), ([17, 17], [-2147483648, 2147483647]))"
        }
      }
    ], 
    "assign5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assign5", 
          "start0"
        ], 
        "values": {
          "i": "[12, 12]", 
          "r": "(([12, 12], [10, 10]), ([17, 17], [22, 22]))"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "i": "[-2147483648, 2147483647]", 
          "r": "(([-2147483648, 2147483647], [-2147483648, 2147483647]), ([-2147483648, 2147483647], [-2147483648, 2147483647]))"
        }
      }
    ]
  }
}
